lib/tasks/rake_helpers/storage.rb

```diff
@@ -0,0 +1,223 @@
+                 CAST(SPLIT_PART(metadata, '/', 2) AS INTEGER),
+                 id,
+                 created_at
+          FROM active_storage_blobs
+          WHERE metadata LIKE 'in-progress/%';
+      SQL
+
+      connection.prepare("update_active_storage_blobs_metadata_#{name}", <<~SQL)
+        UPDATE active_storage_blobs SET metadata='{}' WHERE metadata LIKE 'in-progress/%';
+      SQL
+
+      puts "Creating Active Storage Blobs for #{name}"
+      connection.exec_prepared("create_active_storage_blobs_#{name}");
+      puts "Creating Active Storage Attachments for #{name}"
+      connection.exec_prepared("create_active_storage_attachments_#{name}");
+      puts "Updating Active Storage blobs metadata for #{name}"
+      connection.exec_prepared("update_active_storage_blobs_metadata_#{name}");
+    end
+  end
+
+  def self.rollback(model)
+    connection = ActiveRecord::Base.connection.raw_connection
+    ATTACHMENTS[model].each do |name|
+      # Delete active storage attachments for the model
+      connection.prepare("delete_active_storage_attachments_#{name}", <<~SQL)
+        DELETE FROM active_storage_attachments WHERE record_type='#{self.models(model)}' AND name='#{name}'
+      SQL
+
+      connection.exec_prepared("delete_active_storage_attachments_#{name}");
+    end
+
+    # Delete blobs no longer linked to an attachment
+    connection.prepare("delete_active_storage_blobs", <<~SQL)
+      DELETE FROM active_storage_blobs
+        WHERE id NOT IN (SELECT blob_id FROM active_storage_attachments)
+    SQL
+    connection.exec_prepared("delete_active_storage_blobs");
+  end
+
+  def self.clear_paperclip_checksums(model)
+    connection = ActiveRecord::Base.connection.raw_connection
+    ATTACHMENTS[model].each do |name|
+      # Clear checksums
+      connection.prepare("clear_paperclip_checksums_#{name}", <<~SQL)
+        UPDATE #{model} SET as_#{name}_checksum=NULL
+        WHERE as_#{name}_checksum IS NOT NULL
+      SQL
+
+      result = connection.exec_prepared("clear_paperclip_checksums_#{name}")
+      puts "Updated #{result.cmd_tuples.to_s.green} rows"
+    end
+  end
+
+  def self.create_dummy_paperclip_files_for(model)
+    if self.models(model).nil?
+      puts "Cannot create dummy files for: #{model}"
+      exit
+    end
+
+    ATTACHMENTS[model].each do |name|
+      records = self.models(model).where.not("#{name}_file_name" => nil)
+
+      bar = self.progress_bar title: name, total: records.count
+
+      records.each do |record|
+        bar.increment
+        filename = File.absolute_path(record.send(name).path)
+        FileUtils.mkdir_p File.dirname(filename)
+        File.open(filename, 'wb') { |file| file.write(SecureRandom.random_bytes(record.send("#{name}_file_size"))) }
+      end
+    end
+  end
+
+  def self.set_paperclip_checksums(records:, model:)
+    bar = self.progress_bar title: ATTACHMENTS[model].join(', '), total: records.count
+
+    records.each do |record|
+      bar.increment
```

Comment:
For information more than anything else, as I love the progress bar, but the `bar.increment` carries a speed cost.

I tried using batches and only increment every thousand record and it reduced the time by approx 8.5% 7:55 to 7:16 - this was only tested locally on messages relation.

eg.
```
    relation.in_batches do |batch_relation|
      batch_relation.count.times { bar.increment }
      batch_relation.each do |record|
        record.populate_checksum
        record.save
      end
    end
```